### PR TITLE
Add G-Fork infrastructure

### PR DESCRIPTION
**Pull request by palango**
### Description

Add the basic infrastructure for the G* hardfork.

### CODE REVIEW

**Comment by palango** [comment]
Wouldn't it make more sense to to that in the PR that adds the new instruction?

**Comment by karlb** [comment]
> Wouldn't it make more sense to to that in the PR that adds the new instruction?

If it was a single PR, yes. But since we touch `DIFFICULTY`, `RANDOM`, `PREVRANDAO`, `GASLIMIT` and `BASE_FEE`, I assume there will be multiple PRs.

**Comment by palango** [comment]
@karlb Added the new instruction set.

File: cmd/mycelo/genesis_commands.go [hunk 1]
```diff
@@ -167,6 +167,15 @@ func envFromTemplate(ctx *cli.Context, workdir string) (*env.Environment, *genes
 		}
```
Comment (eelanagaraj) [inline cmd/mycelo/genesis_commands.go:167]: Any reason not to add the CLI flag for `forks.gFork` too?

Comment (palango) [reply cmd/mycelo/genesis_commands.go:167]: Do you mean this below or something else?
https://github.com/celo-org/celo-blockchain/blob/f75b4a3b3f1068d4672b7c8791e4444cb4810331/cmd/mycelo/genesis_commands.go#L170-L178

Comment (palango) [reply cmd/mycelo/genesis_commands.go:167]: The flag definition was actually missing :)

**Review by karlb — APPROVED** [review]
I'm not used to handling forks, but it looks reasonable to me.

**Pull request merged**